src/App.tsx

```diff
@@ -1,40 +1,38 @@
-import { useEffect, useState } from "react";
+import { useState } from "react";
 import TransferButtons from "./TransferButtons";
 import List from "./List";
 import {
-  calculateWordCount,
-  debounce,
   intersection,
   not,
 } from "./utils";
+import { ListInput } from "./components/list-input/ListInput";
+import './App.css';
 
-function App() {
-  const [inputValue, setInputValue] = useState("");
-  const [currWordCount, setCurrWordCount] = useState(0);
-  const [firstListItems, setFirstListItems] = useState([
-    "one",
-    "two",
-    "three",
-    "four",
-  ]);
-  const [secondListItems, setSecondListItems] = useState([]);
+export type ListItem =  { 
+  id: string; 
+  name: string; 
+};
+
+const initialListItems: Array<ListItem> = [
+  { id: "1", name: "item 1" },
+  { id: "2", name: "item 2" },
+]
 
-  const [checkedItems, setChecktedItems] = useState([]);
+function App() {
+  const [firstListItems, setFirstListItems] = useState<Array<ListItem>>(initialListItems);
+  const [secondListItems, setSecondListItems] = useState<Array<ListItem>>([]);
+  const [checkedItems, setChecktedItems] = useState<Array<ListItem>>([]);
+  
   const leftCheckedItems = intersection(checkedItems, firstListItems);
   const rightCheckedItems = intersection(checkedItems, secondListItems);
 
-  function handleCheck(item) {
-    const currIdx = checkedItems.indexOf(item);
-    const checkedItemsToUpdate = [...checkedItems];
-
-    if (currIdx === -1) {
-      //does not found
-      checkedItemsToUpdate.push(item);
-    } else {
-      checkedItemsToUpdate.splice(currIdx, 1);
+  function handleCheck(item: ListItem, checked: boolean) {
+    if(checked){
+      setChecktedItems((prev) => [...prev, item]);
+    }
+    else {
+      setChecktedItems((prev) => prev.filter((prevItem) => prevItem.id !== item.id));
```

Comment:
Here, you don't need to find the item itself, you just need to know if it is checked or not. This information is available in onChange event of checkbox. If checked, you add it to the checkedItems list, if not checked you filter it out. 

Important note:
React state must be set with `setState((prev) => prev + newValue)`
Because, the prev gives you the most recent value.